### PR TITLE
fixed: Music playback from e.g. the library album node was still brok…

### PR DESCRIPTION
**Pull request by arnova**
…en due to an incomplete fix in PR8724

### CODE REVIEW

**Comment by arnova** [comment]
Note that I'm not sure whether the modifying of the CFileItemPtr there is correct. Perhaps we should use a copy?

**Comment by Montellese** [comment]
I don't know the whole call path but if the items come from the GUI then you must not modify their path because other logic relies on that path.

**Comment by arnova** [comment]
@Montellese : Indeed, that's why I had a feeling it was wrong. I've changed this now. Please review.

@ronie: Please test

jenkins build this please

**Comment by ronie** [comment]
great, thx!
ABBA's blasting out of my speakers again :-)

**Comment by Razzeee** [comment]
jenkins build this please

**Comment by DaveTBlake** [comment]
Music playback from album node (when cached?) was broken by #8724 that fixed playback of album from JSON (the "still" in the title is a bit misleading).

Needs backport to Jarvis, or #8756 modifying before it is merged and Jarvis released

**Comment by asavah** [comment]
This PR has unwanted sideffects.
See http://forum.kodi.tv/showthread.php?tid=250817&pid=2207938#pid2207938
http://forum.kodi.tv/showthread.php?tid=250817&pid=2207965#pid2207965

**Comment by MilhouseVH** [comment]
This PR is also responsible for addons failing to load resources: http://forum.kodi.tv/showthread.php?tid=250817&pid=2207966#pid2207966

**Comment by MilhouseVH** [comment]
The Info display during playback is not working with this PR.

http://forum.kodi.tv/showthread.php?tid=250817&pid=2208102#pid2208102

The following appears in the log when this PR is used:
```
04:57:29 27077.593750 T:1964356480   DEBUG: OnKey: i (0xf049) pressed, action is Info
04:57:29 27077.593750 T:1964356480   DEBUG: ------ Window Init (DialogFullScreenInfo.xml) ------
04:57:29 27077.595703 T:1964356480    INFO: Loading skin file: DialogFullScreenInfo.xml, load type: KEEP_IN_MEMORY
04:57:29 27077.595703 T:1964356480   ERROR: unable to load:/usr/share/kodi/addons/skin.confluence/720p/DialogFullScreenInfo.xml, Line 0
                                            Failed to open file
04:57:29 27077.595703 T:1964356480   DEBUG: ------ Window Deinit (DialogFullScreenInfo.xml) ------
```

**Comment by arnova** [comment]
Another regression. *Sigh*. Why didn't we run into this design-flaw in our dircache ages before? Honestly I have no idea atm why this PR is causing those issues. Need to look into it (reproduce it), probably this weekend. Anyone able to debug this and/or has an idea what's going on?

File: xbmc/filesystem/DirectoryCache.cpp [hunk 1]
```diff
@@ -202,7 +201,15 @@ bool CDirectoryCache::FileExists(const std::string& strFile, bool& bInCache)
...
     m_cacheHits++;
 #endif
-    return (URIUtils::PathEquals(strPath, storedPath) || dir->m_Items->Contains(strFile2));
+    if (URIUtils::PathEquals(strPath, storedPath))
+      return true;
+
+    // Look for file in this directory but strip off url options
+    for (int i = 0; i < dir->m_Items->Size(); i++)
+    {
+      if (strFile2 == CURL(dir->m_Items->Get(i)->GetPath()).GetWithoutOptions());
```
Comment (arnova) [inline xbmc/filesystem/DirectoryCache.cpp]: f*****-ing semicolon at eol is the problem (typo).

**Comment by arnova** [comment]
@MilhouseVH / @ronie / @DaveTBlake : Fixed the PR and also properly re-implemented fastlookup. Please test.

File: xbmc/filesystem/DirectoryCache.cpp (hunk 1 above)
Comment (fritsch) [inline xbmc/filesystem/DirectoryCache.cpp]: @arnova yeah review failed - such things happens :-) thanks for spotting it.

**Comment by MilhouseVH** [comment]
The regressions reported by myself and @asavah all seem to be fixed by the latest updates - many thanks!

**Comment by fritsch** [comment]
jenkins build and merge

**Comment by arnova** [comment]
@MartijnKaijser : Note that I've already backported the fixes to the Jarvis PR as well

**Comment by arnova** [comment]
@fritsch: I think there's still one remaining issue with fastlookup. Can be abort the auto merge somehow?

**Comment by fritsch** [comment]
@arnova yes - one moment.

**Comment by fritsch** [comment]
@arnova just push something in the next 5 minutes :-)

**Comment by arnova** [comment]
@MilhouseVH / @ronie / @DaveTBlake : I've updated it. Mind testing it once again?

**Comment by arnova** [comment]
@fritsch: Too late :/

**Comment by fritsch** [comment]
@arnova sorry - I could not stop it - could you please rebase and append?

**Comment by arnova** [comment]
Will do tonight. Obviously needs some more testing again.

**Comment by arnova** [comment]
@MilhouseVH / @ronie / @DaveTBlake : Mind giving current master another try? Just pushed another fix.

**Comment by MilhouseVH** [comment]
Yep, all working OK. :)

**Comment by arnova** [comment]
@MilhouseVH : You tested with current master?

**Comment by MilhouseVH** [comment]
Yes, 77187022e9d60aca60f5e67702a5b0d9d66f06f0

**Comment by Voyager1** [comment]
there are issues with Videos > Files... certain folders not showing up any longer until I rename them.

**Comment by arnova** [comment]
@Voyager1 : You sure it's the commit in this PR causing it? I can't reproduce it here.

**Comment by Voyager1** [comment]
@arnova - sorry for the noise, it's not this PR.

**Comment by DaveTBlake** [comment]
Sorry that this fix has been so painful @arnova, no idea why we didn't hit any of this before the cache limit change, thanks for sticking with it.

I am a little unclear if we have caught up back-porting to Jarvis. Is there a PR for that?

**Pull request merged**